presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkShuffleWriteDescriptor.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import org.apache.spark.shuffle.ShuffleHandle;
+
+public class PrestoSparkShuffleWriteDescriptor
```

Comment:
Looks like write descriptor has no extra stuff. Can we consolidate read/write descriptor to just be PrestoSparkShuffleDescriptor for simplicity? If we need to know if its for read or write we can have an enum to help.

Reply:
The reason for not consolidate read/write to one object is some of the fields only exists for read (partition sizes, partition etc), we can make all of them Optional but that will be make access this field more verbose (have to check if the optional present and then call the get to get the real values), so I'd prefer to keep them separate.